pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -42,58 +44,69 @@ import (
 	samplescheme "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned/scheme"
 	informers "github.com/xmudrii/etcdproxy-controller/pkg/client/informers/externalversions/etcd/v1alpha1"
 	listers "github.com/xmudrii/etcdproxy-controller/pkg/client/listers/etcd/v1alpha1"
+	"github.com/xmudrii/etcdproxy-controller/pkg/utility"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-const controllerAgentName = "etcdstorage-controller"
+const controllerAgentName = "etcdproxy-controller"
 
 const (
 	// SuccessSynced is used as part of the Event 'reason' when a EtcdStorage is synced
 	SuccessSynced = "Synced"
 	// ErrResourceExists is used as part of the Event 'reason' when a EtcdStorage fails
-	// to sync due to a Deployment of the same name already existing.
+	// to sync due to a ReplicaSet of the same name already existing.
 	ErrResourceExists = "ErrResourceExists"
 
 	// MessageResourceExists is the message used for Events when a resource
-	// fails to sync due to a Deployment already existing
+	// fails to sync due to a ReplicaSet already existing
 	MessageResourceExists = "Resource %q already exists and is not managed by EtcdStorage"
 	// MessageResourceSynced is the message used for an Event fired when a EtcdStorage
 	// is synced successfully
 	MessageResourceSynced = "EtcdStorage synced successfully"
 )
 
+type EtcdConnectionInfo struct {
```

Comment:
Doc this type.  it looks weird at first glance, but makes sense when you realize that it is for wiring the information the RS you're going to make.

Reply:
Done.